### ISSUE TITLE
CA2222 creates false-positives for methods with same name but different signature

### ISSUE THREAD

**Issue by x3ntrix** [dotnet/roslyn-analyzers#1565]
#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer

CA2222: Do not decrease inherited member visibility

#### Repro steps

Extend an existing class and create a private method that has the same name as a method with higher visibility from the base class but with different parameters.

```csharp
    public class MyCollection : Collection<string>
    {
        // Causes warning because signature is similar to public Add(string) method from base class
        private void Add(int number)
        {
        }
    }
```

#### Expected behavior
Analyzer does not report a warning because the visibility of the existing method is not decreased.  The method from the extended class and from the base class have different signatures. According to [MSDN ](https://msdn.microsoft.com/en-us/library/ms182332.aspx) the warning should only occur for a private method that has the same signature as a public method from the base class. 

#### Actual behavior
Analyzer reports warning for method with same name but different signature.

**Comment by sharwell** [comment]
This is no longer applicable. CA2222 was removed in #1554.
